hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -1702,7 +1687,19 @@ public OzoneKeyDetails getS3KeyDetails(String bucketName, String keyName,
             partNumber)
         .collect(Collectors.toList());
     keyInfo.updateLocationInfoList(filteredKeyLocationInfo, false);
-    return getOzoneKeyDetails(keyInfo, true);
+    keyInfo.setDataSize(getTotalBytesRead(keyInfo));
+    return getOzoneKeyDetails(keyInfo);
+  }
+
+  private long getTotalBytesRead(OmKeyInfo keyInfo) throws IOException {
+    OzoneInputStream keyContent = getInputStreamWithRetryFunction(keyInfo);
+    byte[] buffer = new byte[4096];
+    int bytesRead;
+    long totalBytesRead = 0;
+    while ((bytesRead = keyContent.read(buffer)) != -1) {
+      totalBytesRead += bytesRead;
+    }
+    return totalBytesRead;
```

Comment:
Does this mean client reads data for the specified part just to calculate size for `ContentLength`?

Reply:
That's right, I was not able to see any existing way in which the OmKeyInfo#dataSize could dynamically get updated based on the updated `OmKeyLocationInfo` list.

I am open to suggestions if there is a better way to handle this!

Reply:
As discussed offline with @kerneltime, I think we can have the block lengths summed up for a given `partNumber` instead - I will update the changes accordingly, thanks!